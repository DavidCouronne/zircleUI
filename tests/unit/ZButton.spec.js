import { shallowMount, createLocalVue } from '@vue/test-utils'
import zbutton from '@/components/z-button.vue'
import zircle from '@/index'
const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zbutton, {
  propsData: {
    color: 'success',
    imagesrc: './images/test.png',
    label: 'Test label prop'
  },
  provide: function () {
    return {
      view: 'test'
    }
  },
  computed: {
    position: function () {
      return {X: 0, Y: 0, scale: 1, Xi: 0, Yi: 0, scalei: 1}
    }
  },
  slots: {
    default: '<p> lorem text </p>',
    imagesrc: './images/dummy.png',
    label: 'Test label slot',
    extension: '<div></div>'
  },
  localVue
})
describe('z-button.vue', () => {
  it('Renders props when passed', () => {
    expect(wrapper.vm.color).toEqual('success')
    expect(wrapper.vm.imagesrc).toEqual('./images/test.png')
    expect(wrapper.vm.label).toEqual('Test label prop')
  })
  it('Has the expected html structure', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
