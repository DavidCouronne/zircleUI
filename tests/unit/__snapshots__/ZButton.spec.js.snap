// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-button.vue Has the expected html structure 1`] = `
<div
  class="zui disc button success"
  style="width: 30px; height: 30px; margin: -15px 0px 0px -15px; transform: translate3d(0px, 0px, 0px);"
  title="z-button"
  type="button"
>
  <div
    class="z-pulse"
  />
   
  <section
    class="label"
  >
    
    Test label prop 
    
    <!---->
  </section>
   
  <div
    class="z-content"
  >
    <img
      height="100%"
      src="./images/test.png"
      width="100%"
    />
     
    <!---->
  </div>
   
  <div
    class="z-content"
  >
    <span
      class="overflow"
    >
      <p>
         lorem text 
      </p>
    </span>
  </div>
   
  <div />
</div>
`;
